src/io/syncsocket.c

```diff
@@ -228,27 +230,62 @@ static void gc_free(MVMThreadContext *tc, MVMObject *h, void *d) {
     MVM_free(data);
 }
 
+static size_t get_struct_size_for_family(sa_family_t family) {
+    switch (family) {
+        case AF_INET6:
+            return sizeof(struct sockaddr_in6);
+        case AF_INET:
+            return sizeof(struct sockaddr_in);
+#ifndef _WIN32
+        case AF_UNIX:
+            return sizeof(struct sockaddr_un);
+#endif
+        default:
+            return sizeof(struct sockaddr);
+    }
+}
+
 /* Actually, it may return sockaddr_in6 as well; it's not a problem for us, because we just
  * pass is straight to uv, and the first thing it does is it looks at the address family,
  * but it's a thing to remember if someone feels like peeking inside the returned struct. */
```

Comment:
Can you add a comment to explain the newly added flags? Thanks.

Reply:
Is this okay?